test/mock-pubsub.test.ts

```diff
@@ -0,0 +1,263 @@
+              const topic = pubsub.topic('non-existing');
+              expect(topic.name).toBe(
+                `projects/${projectId}/topics/non-existing`,
+              );
+
+              expect(topic.publish).toEqual(expect.any(Function));
+              expect(topic.publishMessage).toEqual(expect.any(Function));
+
+              const [existingTopics] = await pubsub.getTopics();
+              expect(existingTopics).toHaveLength(0);
+            });
+          });
+        });
+
+        describe('pubsub.getSubscriptions()', () => {
+          it('should return list of existing subscriptions for given PubSub instance', async () => {
+            const otherProjectId = 'another-project-id';
+            const otherPubsub = new PubSubClass({
+              projectId: otherProjectId,
+            });
+            await clearPubSubInstance(otherPubsub);
+
+            const [topic] = await pubsub.createTopic('topic1');
+            const [otherProjectTopic] = await otherPubsub.createTopic('topic1');
+
+            await Promise.all([
+              await topic.createSubscription('sub1'),
+              await topic.createSubscription('sub2'),
+              await otherProjectTopic.createSubscription('sub1'),
+              await otherProjectTopic.createSubscription('sub2'),
+            ]);
+
+            // pubsub instance 1
+            {
+              const [subscriptions] = await pubsub.getSubscriptions();
+              const subNames = subscriptions.map((s) => s.name);
+
+              expect(subNames).toEqual([
+                `projects/${projectId}/subscriptions/sub1`,
+                `projects/${projectId}/subscriptions/sub2`,
+              ]);
+            }
+
+            // pubsub instance 2
+            {
+              const [subscriptions] = await otherPubsub.getSubscriptions();
+              const subNames = subscriptions.map((s) => s.name);
+
+              expect(subNames).toEqual([
+                `projects/${otherProjectId}/subscriptions/sub1`,
+                `projects/${otherProjectId}/subscriptions/sub2`,
+              ]);
+            }
+          });
+        });
+
+        describe('pubsub.subscription()', () => {
+          it('should retrieve previously created subscription by name', async () => {
+            const [topic] = await pubsub.createTopic('topic');
+            await topic.createSubscription('sub1');
+            await topic.createSubscription('sub2');
+
+            const sub1 = pubsub.subscription('sub1');
+            const sub2 = pubsub.subscription(
+              `projects/${projectId}/subscriptions/sub2`,
+            );
+
+            expect(sub1.name).toBe(`projects/${projectId}/subscriptions/sub1`);
+            expect(sub2.name).toBe(`projects/${projectId}/subscriptions/sub2`);
+          });
+
+          // Execute once this PR gets merged
+          // https://github.com/mkls/mock-google-cloud-pubsub/pull/3
+          describe.skip('non existing subscription', () => {
```

Comment:
Enable tes once
https://github.com/mkls/mock-google-cloud-pubsub/pull/31
is merged